js/parts/AreaSeries.js

```diff
@@ -149,6 +149,7 @@ seriesType('area', 'line',
      * @sample {highcharts} highcharts/plotoptions/area-threshold-infinity/
      *         A threshold of Infinity
      *
+     * @type    {number|null}
```

Comment:
It is not specific to area series and its children, it also applies to [column series](https://api.highcharts.com/highcharts/series.column.threshold).

In `line` series however, it doesn't seem to have an effect, because the Y axis doesn't react to any `threshold` setting. So for line series it seems that `number` is the only type that makes sense. Although for consistency we could allow `null` there too, it will affect the `negativeColor` the same way as setting it to 0.